fix the test failed

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

```
def test_sse_message_id_coercion():
        """Test that string message IDs that look like integers are parsed as integers.
    
        See <https://github.com/modelcontextprotocol/python-sdk/pull/851> for more details.
        """
        json_message = '{"jsonrpc": "2.0", "id": "123", "method": "ping", "params": null}'
        msg = types.JSONRPCMessage.model_validate_json(json_message)
        expected = types.JSONRPCMessage(root=types.JSONRPCRequest(method="ping", jsonrpc="2.0", id=123))
    
        # Check if both are JSONRPCRequest instances
>       assert isinstance(msg.root, types.JSONRPCRequest)
E       AssertionError: assert False
E        +  where False = isinstance(JSONRPCNotification(method='ping', params=None, jsonrpc='2.0', id='123'), <class 'core.mcp.types.JSONRPCRequest'>)
E        +    where JSONRPCNotification(method='ping', params=None, jsonrpc='2.0', id='123') = JSONRPCMessage(root=JSONRPCNotification(method='ping', params=None, jsonrpc='2.0', id='123')).root
E        +    and   <class 'core.mcp.types.JSONRPCRequest'> = types.JSONRPCRequest
```

it can be basically determined that this is an issue arising from changes in Pydantic v2's union type matching behavior. But more precisely, it's the result of these factors (stacking together):

- JSONRPCMessage is a plain union: JSONRPCRequest | JSONRPCNotification | JSONRPCResponse | JSONRPCError
- JSONRPCNotification has extra="allow" enabled, so although it doesn't declare an id field, it can still accept input like {"id": "123", "method": "ping"}.
- id="123" in JSONRPCRequest goes through RequestId = Annotated[int | str, Field(union_mode="left_to_right")] and gets coerced into 123.
- Newer versions of Pydantic prefer the union branch that is more precise / involves less coercion, so it may ultimately classify the input as JSONRPCNotification instead of JSONRPCRequest.

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

--- a/api/core/mcp/types.py
+++ b/api/core/mcp/types.py
@@ -2,7 +2,7 @@ from collections.abc import Callable
 from dataclasses import dataclass
 from typing import Annotated, Any, Literal
 
-from pydantic import BaseModel, ConfigDict, Field, FileUrl, RootModel
+from pydantic import BaseModel, ConfigDict, Field, FileUrl, RootModel, model_validator
 from pydantic.networks import AnyUrl, UrlConstraints
 
 """
@@ -173,7 +173,21 @@ class JSONRPCError(BaseModel):
 
 
 class JSONRPCMessage(RootModel[JSONRPCRequest | JSONRPCNotification | JSONRPCResponse | JSONRPCError]):
-    pass
+    @model_validator(mode="before")
+    @classmethod
+    def _select_message_type(
+        cls, value: Any
+    ) -> JSONRPCRequest | JSONRPCNotification | JSONRPCResponse | JSONRPCError | Any:
+        if isinstance(value, dict):
+            if "result" in value:
+                return JSONRPCResponse.model_validate(value)
+            if "error" in value:
+                return JSONRPCError.model_validate(value)
+            if "method" in value:
+                if "id" in value:
+                    return JSONRPCRequest.model_validate(value)
+                return JSONRPCNotification.model_validate(value)
+        return value
 
 
 class EmptyResult(Result):

--- a/api/tests/unit_tests/core/mcp/client/test_sse.py
+++ b/api/tests/unit_tests/core/mcp/client/test_sse.py
@@ -34,6 +34,17 @@ def test_sse_message_id_coercion():
     assert msg.root.jsonrpc == expected.root.jsonrpc
 
 
+def test_sse_message_without_id_stays_notification():
+    """Test that method messages without an ID still parse as notifications."""
+    json_message = '{"jsonrpc": "2.0", "method": "ping", "params": null}'
+
+    msg = types.JSONRPCMessage.model_validate_json(json_message)
+
+    assert isinstance(msg.root, types.JSONRPCNotification)
+    assert msg.root.method == "ping"
+    assert msg.root.jsonrpc == "2.0"
+
+
 class MockSSEClient:
     """Mock SSE client for testing."""
 

